libcloudforensics/providers/aws/internal/ebs.py

```diff
@@ -305,3 +305,308 @@ def ShareWithAWSAccount(self, aws_account_id: str) -> None:
+          snapshot, volume_name_prefix=volume_name_prefix)
+
+    if len(volume_name) > 255:
+      raise ValueError(
+          'Volume name {0:s} is too long (>255 chars)'.format(volume_name))
+
+    if not tags:
+      tags = {}
+    tags['Name'] = volume_name
+
+    client = self.aws_account.ClientApi(common.EC2_SERVICE)
+    create_volume_args = {
+        'AvailabilityZone': snapshot.availability_zone,
+        'SnapshotId': snapshot.snapshot_id,
+        'TagSpecifications': [common.CreateTags(common.VOLUME, tags)],
+        'VolumeType': volume_type
+    }
+    if kms_key_id:
+      create_volume_args['Encrypted'] = True
+      create_volume_args['KmsKeyId'] = kms_key_id
+    if volume_type == 'io1':
+      # If using the io1 volume type, we must specify Iops, see
+      # https://boto3.amazonaws.com/v1/documentation/api/latest/reference/
+      # services/ec2.html#EC2.Client.create_volume. io1 volumes allow for a
+      # ratio of 50 IOPS per 1 GiB.
+      create_volume_args['Iops'] = self.aws_account.ResourceApi(
+          common.EC2_SERVICE).Snapshot(snapshot.snapshot_id).volume_size * 50
+    try:
+      volume = client.create_volume(**create_volume_args)
+      volume_id = volume['VolumeId']
+      # Wait for volume creation completion
+      client.get_waiter('volume_available').wait(VolumeIds=[volume_id])
+    except (client.exceptions.ClientError,
+            botocore.exceptions.WaiterError) as exception:
+      raise RuntimeError('Could not create volume {0:s} from snapshot '
+                         '{1:s}: {2:s}'.format(volume_name, snapshot.name,
+                                               str(exception)))
+
+    zone = volume['AvailabilityZone']
+    encrypted = volume['Encrypted']
+
+    return AWSVolume(volume_id,
+                     self.aws_account,
+                     self.aws_account.default_region,
+                     zone,
+                     encrypted,
+                     name=volume_name)
+
+  def GetAccountInformation(self, info: str) -> str:
+    """Get information about the AWS account in use.
+
+    If the call succeeds, then the response from the STS API is expected to
+    have the following entries:
+      - UserId
+      - Account
+      - Arn
+    See https://boto3.amazonaws.com/v1/documentation/api/1.9.42/reference/services/sts.html#STS.Client.get_caller_identity for more details. # pylint: disable=line-too-long
+
+    Args:
+      info (str): The account information to retrieve. Must be one of [UserID,
+          Account, Arn]
+    Returns:
+      str: The information requested.
+
+    Raises:
+      KeyError: If the requested information doesn't exist.
+    """
+    account_information = self.aws_account.ClientApi(
+        common.ACCOUNT_SERVICE).get_caller_identity()  # type: Dict[str, str]
+    if not account_information.get(info):
+      raise KeyError('Key must be one of ["UserId", "Account", "Arn"]')
+    return account_information[info]
```

Comment:
I'm not sure how/where this is called but could we not just return all of them? It's only 3 values, could just let the callee filter if necessary and if someone wanted all the info they would currently have to call this 3 times.

Reply:
I agree, maybe it would be better to return a dict with those three keys and let the caller sort out which ones they need.